src/tsickle.ts

```diff
@@ -1084,20 +1084,23 @@ class ExternsWriter extends ClosureRewriter {
               this.writeExternsVariable(name, namespace, '{}');
             }
             if (decl.body) this.visit(decl.body, namespace.concat(name));
-          } break;
-          case ts.SyntaxKind.StringLiteral: {
+            break;
+          case ts.SyntaxKind.StringLiteral:
             // E.g. "declare module 'foo' {" (note the quotes).
             // We still want to emit externs for this module, but
             // Closure doesn't really provide a mechanism for
             // module-scoped externs.  For now, ignore the enclosing
             // namespace (because this is declaring a top-level module)
             // and emit into a fake namespace.
-            namespace = ['tsickle_declare_module'];
-            let name = (decl.name as ts.StringLiteral).text;
+            namespace
+            = ['tsickle_declare_module'];
+            let importName = (decl.name as ts.StringLiteral).text;
+            // Imports are quoted strings, which allow characters like "-".
+            importName = importName.replace(/-/, '_');
```

Comment:
This is in `declare module "foo"`, so I think + isn't common?  But I bet @ is, ugh.